kover-gradle-plugin/src/functionalTest/kotlin/kotlinx/kover/gradle/plugin/test/functional/cases/SourceSetsTests.kt

```diff
@@ -0,0 +1,127 @@
+        val buildResult = gradleBuild.runWithParams(":koverXmlReport")
+
+        gradleBuild.createCheckerContext(buildResult).xmlReport {
+            assertTrue(buildResult.isSuccessful, "Build should be successful")
+            classCounter("kotlinx.kover.examples.sourcesets.MainClass").assertAbsent()
+            classCounter("kotlinx.kover.examples.sourcesets.FooClass").assertAbsent()
+            classCounter("kotlinx.kover.examples.sourcesets.ExtraClass").assertPresent()
+        }
+    }
+
+    @Test
+    fun testInclude() {
+        val template = buildFromTemplate("sourcesets-multi")
+
+        template.kover {
+            currentProject {
+                sources.includedSourceSets.addAll("extra")
+            }
+        }
+
+        val gradleBuild = template.generate()
+        val buildResult = gradleBuild.runWithParams(":koverXmlReport")
+
+        gradleBuild.createCheckerContext(buildResult).xmlReport {
+            assertTrue(buildResult.isSuccessful, "Build should be successful")
+            classCounter("kotlinx.kover.examples.sourcesets.MainClass").assertAbsent()
+            classCounter("kotlinx.kover.examples.sourcesets.FooClass").assertAbsent()
+            classCounter("kotlinx.kover.examples.sourcesets.ExtraClass").assertPresent()
+        }
+    }
+
+    @Test
+    fun testInclude2() {
+        val template = buildFromTemplate("sourcesets-multi")
+
+        template.kover {
+            currentProject {
+                sources.includedSourceSets.addAll("extra", "foo")
+            }
+        }
+
+        val gradleBuild = template.generate()
+        val buildResult = gradleBuild.runWithParams(":koverXmlReport")
+
+        gradleBuild.createCheckerContext(buildResult).xmlReport {
+            assertTrue(buildResult.isSuccessful, "Build should be successful")
+            classCounter("kotlinx.kover.examples.sourcesets.MainClass").assertAbsent()
+            classCounter("kotlinx.kover.examples.sourcesets.FooClass").assertPresent()
+            classCounter("kotlinx.kover.examples.sourcesets.ExtraClass").assertPresent()
+        }
+    }
+
+    @Test
+    fun testMixed() {
+        val template = buildFromTemplate("sourcesets-multi")
+
+        template.kover {
+            currentProject {
+                sources {
+                    includedSourceSets.addAll("extra", "foo")
+                    excludedSourceSets.add("foo")
+                }
+            }
+        }
+
+        val gradleBuild = template.generate()
+        val buildResult = gradleBuild.runWithParams(":koverXmlReport")
+
+        gradleBuild.createCheckerContext(buildResult).xmlReport {
+            assertTrue(buildResult.isSuccessful, "Build should be successful")
+            classCounter("kotlinx.kover.examples.sourcesets.MainClass").assertAbsent()
+            classCounter("kotlinx.kover.examples.sourcesets.FooClass").assertAbsent()
+            classCounter("kotlinx.kover.examples.sourcesets.ExtraClass").assertPresent()
```

Comment:
Add `classCounter("kotlinx.kover.examples.sourcesets.TestClasses")` to this test and all above